Mathlib/NumberTheory/AbelSummation.lean

```diff
@@ -0,0 +1,174 @@
+/-
+Copyright (c) 2024 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.MeasureTheory.Integral.FundThmCalculus
+
+/-!
+# Abel's summation formula
+
+We prove several versions of Abel's summation formula.
```

Comment:
Note that these are only for a sum of complex numbers; I didn't manage to derive the real version. (Is differentiability actually hard when you throw in the coercion from Real to Complex or am I bad at searching the library?)

Maybe these results should be namespaced under `Complex` to leave room for the real ones when we add them

Reply:
I would suggest treating the real and complex version simultaneously using `RCLike`. If you assume `{𝕜 : Type*} [RCLike 𝕜] {f : ℝ → 𝕜} (c : ℕ → 𝕜)` then Xavier's proof works without any change at all.

(Probably it would work with vector-valued functions too, but that would be more awkward to formulate. I managed to get the proof working for `f` valued in an `RCLike 𝕜` and `c` valued in a general normed `𝕜`-vector space, modulo one `sorry` since `IntervalIntegrable.smul_const` doesn't exist; but the asymmetry is annoying, why should it be `c` that is vector-valued and not `f`?)

Reply:
> I would suggest treating the real and complex version simultaneously using `RCLike`. If you assume `{𝕜 : Type*} [RCLike 𝕜] {f : ℝ → 𝕜} (c : ℕ → 𝕜)` then Xavier's proof works without any change at all.

I am always so amazed by Lean when things work that smoothly 😄 

> (Probably it would work with vector-valued functions too, but that would be more awkward to formulate. I managed to get the proof working for f valued in an RCLike 𝕜 and c valued in a general normed 𝕜-vector space, modulo one sorry since IntervalIntegrable.smul_const doesn't exist; but the asymmetry is annoying, why should it be c that is vector-valued and not f?)

Do you think this kind of generalization would be interesting?

Reply:
> Do you think this kind of generalization would be interesting?

I don't have a particular application in mind, no. It's only worth doing if it's essentially free (which I think is not true).

Reply:
Well, it will be for a future refactor then